Асинхронная загрузка MITK файлов

sergey-denisov opened the pull request:
http://jira.samsmu.net:8083/browse/AUT-1804

sergey-denisov committed:
load mitk in threads

AUT-1804

Cynede commented:
```
[ 18%] Building CXX object Modules/Core/CMakeFiles/MitkCore.dir/src/DataManagement/mitkEnumerationProperty.cpp.o
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/4.9.3/include/g++-v4/shared_mutex:35:0,
                 from /var/lib/jenkins/home/jobs/MITK-PR/workspace/Modules/Core/src/DataManagement/mitkEnumerationProperty.cpp:19:
/usr/lib/gcc/x86_64-pc-linux-gnu/4.9.3/include/g++-v4/bits/c++14_warning.h:32:2: error: #error This file requires compiler and library support for the forthcoming ISO C++ 2014 standard. This support is currently experimental, and must be enabled with the -std=c++1y or -std=gnu++1y compiler options.
 #error This file requires compiler and library support for the forthcoming \
  ^
/var/lib/jenkins/home/jobs/MITK-PR/workspace/Modules/Core/src/DataManagement/mitkEnumerationProperty.cpp:24:6: error: ‘shared_mutex’ in namespace ‘std’ does not name a type
 std::shared_mutex s_IdMapForClassNameGuard;
      ^
/var/lib/jenkins/home/jobs/MITK-PR/workspace/Modules/Core/src/DataManagement/mitkEnumerationProperty.cpp:25:6: error: ‘shared_mutex’ in namespace ‘std’ does not name a type
 std::shared_mutex s_StringMapForClassNameGuard;
```

интересно на 5.4 соберётся или нет

Cynede committed:
Replace std::shared_mutex with boost realization
std realization was stabilised in C++17 standard
it was added as draft c++1y but then it was renamed to std::shared_timed_mutex
